flink-state-backends/flink-statebackend-rocksdb/src/test/java/org/apache/flink/contrib/streaming/state/RocksDBStateBackendConfigTest.java

```diff
@@ -359,6 +361,50 @@ public void testDbPathRelativePaths() throws Exception {
         rocksDbBackend.setDbStoragePath("relative/path");
     }
 
+    @Test
+    public void testCleanRelocatedDbLogs() throws Exception {
+        final File folder = tempFolder.newFolder();
+        final File relocatedDBLogDir = tempFolder.newFolder("db_logs");
+        final File logFile = new File(relocatedDBLogDir, "taskManager.log");
+        Files.createFile(logFile.toPath());
+        System.setProperty("log.file", logFile.getAbsolutePath());
+
+        final String dbStoragePath = new Path(folder.toURI().toString()).toString();
+        final EmbeddedRocksDBStateBackend rocksDbBackend = new EmbeddedRocksDBStateBackend();
+        rocksDbBackend.setDbStoragePath(dbStoragePath);
+
+        final MockEnvironment env = getMockEnvironment(tempFolder.newFolder());
+        RocksDBKeyedStateBackend<Integer> keyedBackend =
+                createKeyedStateBackend(rocksDbBackend, env, IntSerializer.INSTANCE);
+        // clear unused file
+        FileUtils.deleteFileOrDirectory(logFile);
+
+        File instanceBasePath = keyedBackend.getInstanceBasePath();
+        File instanceRocksDBPath =
+                RocksDBKeyedStateBackendBuilder.getInstanceRocksDBPath(instanceBasePath);
+
+        // avoid tests without relocate.
+        Assume.assumeTrue(instanceRocksDBPath.getAbsolutePath().length() <= 255 - "_LOG".length());
+
+        String relocatedDbLogPrefix =
+                RocksDBResourceContainer.resolveRelocatedDbLogPrefix(
+                        instanceRocksDBPath.getAbsolutePath());
+        java.nio.file.Path[] relocatedDbLogs;
+        try {
+            relocatedDbLogs = FileUtils.listDirectory(relocatedDBLogDir.toPath());
+            assertTrue(relocatedDbLogs[0].getFileName().startsWith(relocatedDbLogPrefix));
+            // add a rolled log file
+            Files.createTempFile(relocatedDBLogDir.toPath(), relocatedDbLogPrefix, ".suffix");
```

Comment:
Could we let rocksdb create LOG file and calling the close to clean it?

If so, we don't need to call the `resolveRelocatedDbLogPrefix`. Current test just tests create an file based on `resolveRelocatedDbLogPrefix`, and clean it based on `resolveRelocatedDbLogPrefix`.

If `resolveRelocatedDbLogPrefix` has some bugs or its rule isn't same with rocksdb side. This test cannot cover it.

Also, if we let rocksdb create LOG file, current test will fail if rocksdb log file rule is changed in the future.

WDYT?

Reply:
Resolved.